File: src/my_cpp_pkg/src/hw_status_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "my_robot_interfaces/msg/hardware_status.hpp"

class HardwareStatusPublisherNode : public rclcpp::Node
{
public:
    HardwareStatusPublisherNode() : Node("hardwareStatusPublisher")
    {
        pub_ = this->create_publisher<my_robot_interfaces::msg::HardwareStatus>(
            "hardware_status", 10);
        timer_ = this->create_wall_timer(std::chrono::seconds(1),
                                         std::bind(&HardwareStatusPublisherNode::publish_hw_status, this));
        RCLCPP_INFO(this->get_logger(), "hardwareStatusPublisher has been started");
    }

private:
    void publish_hw_status()
    {
        auto msg = my_robot_interfaces::msg::HardwareStatus();
        msg.temperature = 77;
        msg.are_motors_ready = false;
        msg.debug_message = "Error Overheat!";
        pub_->publish(msg);
    }
    rclcpp::Publisher<my_robot_interfaces::msg::HardwareStatus>::SharedPtr pub_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<HardwareStatusPublisherNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/hw_status_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from my_robot_interfaces.msg import HardwareStatus


class HardwareStatusPublisherNode(Node):
   def __init__(self):
      super().__init__("hardware_status_publisher")
      self.pub_ = self.create_publisher(HardwareStatus, "hardware_status", 10)
      self.timer_ = self.create_timer(1.0,self.publish_hw_status)


      self.get_logger().info("hardware_status_publisher has been started")

   def publish_hw_status(self):
      msg=HardwareStatus()
      msg.temperature = 40
      msg.are_motors_ready = True
      msg.debug_message = "ok"
      self.pub_.publish(msg)

def main(args=None):
   rclpy.init(args=args)
   node = HardwareStatusPublisherNode()
   rclpy.spin(node)
   rclpy.shutdown()

if __name__ == "__main__":
   main()